File: app/controllers/iop/transfer/api/ExternalReservationHandler.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package controllers.iop.transfer.api;

import com.google.inject.ImplementedBy;
import controllers.iop.transfer.impl.ExternalReservationHandlerImpl;
import java.util.Optional;
import java.util.concurrent.CompletionStage;
import models.enrolment.Reservation;
import models.user.User;
import play.mvc.Result;

@ImplementedBy(ExternalReservationHandlerImpl.class)
public interface ExternalReservationHandler {
    CompletionStage<Result> removeReservation(Reservation reservation, User user, String msg);
    CompletionStage<Optional<Integer>> removeExternalReservation(Reservation reservation);
}

File: app/features/iop/transfer/api/ExternalReservationHandler.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package features.iop.transfer.api

import com.google.inject.ImplementedBy
import features.iop.transfer.impl.ExternalReservationHandlerImpl
import models.enrolment.Reservation
import models.user.User
import play.api.mvc.Result

import scala.concurrent.Future

@ImplementedBy(classOf[ExternalReservationHandlerImpl])
trait ExternalReservationHandler:
  def removeReservation(reservation: Reservation, user: User, msg: String): Future[Result]
  def removeExternalReservation(reservation: Reservation): Future[Option[Int]]
